src/Datadog.Trace/Util/DomainMetadata.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Diagnostics;
+
+namespace Datadog.Trace.Util
+{
+    internal static class DomainMetadata
```

Comment:
I understand why you're adding this, but I also think it's a good case for a documentation header that explains it.